src/librustc_lint/builtin.rs

```diff
@@ -641,9 +641,29 @@ impl LintPass for UnusedAttributes {
             }
         }
 
+        let plugin_attributes = cx.sess().plugin_attributes.borrow_mut();
+        for &(ref name, ty) in plugin_attributes.iter() {
+            match ty {
+                AttributeType::Whitelisted if attr.check_name(&*name) => {
+                    break;
```

Comment:
Also, can't this be `if ty == AttributeType::Whitelisted && attr.check_name(&*name) { break }`?